code/modules/assembly/bomb.dm

```diff
@@ -119,7 +119,7 @@
 	var/strength = 1
 
 	var/turf/ground_zero = get_turf(loc)
-	loc = null
+	moveToNullspace()
```

Comment:
you should just reorder this to calculate the explosion strength, then qdel itself, then trigger the explosion